src/network/networkd-can.c

```diff
@@ -7,10 +7,49 @@
 #include "networkd-can.h"
 #include "networkd-link.h"
 #include "networkd-manager.h"
+#include "parse-util.h"
 #include "string-util.h"
 
 #define CAN_TERMINATION_OHM_VALUE 120
 
+int config_parse_can_bitrate(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint32_t *br = (uint32_t*) data;
+        uint64_t sz;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = parse_size(rvalue, 1000, &sz);
+
+        /* Linux uses __u32 for bitrates, so the value should not exceed that. */
+        if (sz == 0 || sz > UINT32_MAX)
```

Comment:
if parse_size() failed, sz is not initialized, and we shouldn't touch uninitialized variables. please rewrite as:

```c
if (r < 0) {
        log_syntax(…);
        return 0;
} 
if (sz <= 0 || sz > UINT32_MAX) {
        log_syntax(…, "Bit rate out of permitted range 0…4294967295: …", …);
        return 0;
}
```

i.e. generate an explicit error message for this, indicating the permitted range